pkg/install/stack/kubermatic/stack.go

```diff
@@ -0,0 +1,274 @@
+	RawKubermaticConfiguration *unstructured.Unstructured
+	ForceHelmReleaseUpgrade    bool
+}
+
+func Deploy(ctx context.Context, logger *logrus.Entry, kubeClient ctrlruntimeclient.Client, helmClient helm.Client, opt Options) error {
+	if err := deployStorageClass(ctx, logger, kubeClient); err != nil {
+		return fmt.Errorf("failed to deploy StorageClass: %v", err)
+	}
+
+	if err := deployNginxIngressController(ctx, logger, kubeClient, helmClient, opt); err != nil {
+		return fmt.Errorf("failed to deploy nginx-ingress-controller: %v", err)
+	}
+
+	if err := deployCertManager(ctx, logger, kubeClient, helmClient, opt); err != nil {
+		return fmt.Errorf("failed to deploy cert-manager: %v", err)
+	}
+
+	if err := deployDex(ctx, logger, kubeClient, helmClient, opt); err != nil {
+		return fmt.Errorf("failed to deploy Dex: %v", err)
+	}
+
+	if err := deployKubermaticOperator(ctx, logger, kubeClient, helmClient, opt); err != nil {
+		return fmt.Errorf("failed to deploy Kubermatic Operator: %v", err)
+	}
+
+	if err := applyKubermaticConfiguration(ctx, logger, kubeClient, opt); err != nil {
+		return fmt.Errorf("failed to apply Kubermatic Configuration: %v", err)
+	}
+
+	return nil
+}
+
+func deployStorageClass(ctx context.Context, logger *logrus.Entry, kubeClient ctrlruntimeclient.Client) error {
+	logger.Infof("💾 Deploying %s StorageClass…", StorageClassName)
+
+	cls := storagev1.StorageClass{}
+	key := types.NamespacedName{Name: StorageClassName}
+
+	err := kubeClient.Get(ctx, key, &cls)
+
+	// storage class exists already
+	if err == nil {
+		logger.Info("✅ StorageClass exists, nothing to do.")
+		return nil
+	}
+
+	if !kerrors.IsNotFound(err) {
+		return fmt.Errorf("failed to check for StorageClass %s: %v", StorageClassName, err)
+	}
+
+	sc := storageClassForProvider(StorageClassName, "gke")
+	if sc == nil {
+		return fmt.Errorf("cannot automatically create StorageClass %s for this cloud provider, please create it manually", StorageClassName)
+	}
+
+	if err := kubeClient.Create(ctx, sc); err != nil {
+		return fmt.Errorf("failed to create StorageClass %s: %v", StorageClassName, err)
+	}
+
+	logger.Info("✅ Success.")
+
+	return nil
+}
+
+func deployNginxIngressController(ctx context.Context, logger *logrus.Entry, kubeClient ctrlruntimeclient.Client, helmClient helm.Client, opt Options) error {
+	logger.Info("📦 Deploying nginx-ingress-controller…")
+	sublogger := log.Prefix(logger, "   ")
+
+	chart, err := helm.LoadChart("charts/nginx-ingress-controller")
+	if err != nil {
+		return fmt.Errorf("failed to load Helm chart: %v", err)
+	}
+
+	if err := util.EnsureNamespace(ctx, sublogger, kubeClient, NginxIngressControllerNamespace); err != nil {
+		return fmt.Errorf("failed to create namespace: %v", err)
+	}
+
+	release, err := util.CheckHelmRelease(ctx, sublogger, helmClient, NginxIngressControllerNamespace, NginxIngressControllerReleaseName)
+	if err != nil {
+		return fmt.Errorf("failed to check to Helm release: %v", err)
```

Comment:
CheckHelmRelease is giving back a non nil error value causing the installation to abort. This is the case for new installation as the Helm chart doesn't exist thus leading to a failure.

Reply:
You probably maybe don't have the Helm binary in the same symlinked location that I do.

Reply:
@xrstf can't we embed Helm in the installer in this case? 
I modified the path to my Helm3 binary and it works now but for production purposes, I don't think this might be the best way to go.

Reply:
The plan is to ship the installer as a Docker image with a Helm binary and the Helm charts included. When the installer was started, embedding Helm was a PITA and I have since not looked again, but it's the reason why we have a very simple Helm interface in the code, so we can easily change the implementation *later*.

Reply:
And the name for the current Helm 3 binary in the installer is mostly because I have roughly 8 different Helm binaries on my machine. ^^

Reply:
I also mentioned this in the proposal:

> Helm is not easy to embed, so I chose to just shell out to calling a Helm binary. As the installer is delivered in a Docker image (so it can bundle the Helm charts), this seemed a simple choice. The interaction with Helm goes through a super simple interface, so it would be possible to swap out the implementation later.

;-)

Reply:
Noted